main/calculations.py

```diff
@@ -0,0 +1,147 @@
+import requests
+import json
+from datetime import datetime
+from constance import config
+from proxy_requests.proxy_requests import ProxyRequests
+
+from django.core.mail import send_mail
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib import messages
+
+from .models import SiteToCheck
+
+
+def send_email(message, email):
+    subject = 'Errors on my pages'
+    message = message
+    from_email = settings.EMAIL_ADDRESS
+    to_email = email
+    send_mail(
+        subject,
+        message,
+        from_email,
+        [to_email],
+        fail_silently=False,
+    )
+
+
+def my_cron_job():
+    users = User.objects.all()
+    for user in users:
+        sites = SiteToCheck.objects.filter(user=user)
+        output = ''
+        for site in sites:
+            if 'bad_data' in SiteDownChecker(site, user).status():
+                output += f'{site} - ERROR\n'
+            elif site.last_status != 200:
+                output += f'{site} - last status: {site.last_status}'
+        if len(output) > 0:
+            send_email(output, user.email)
+
+
+def modify_email(request):
+    response_json = json.dumps(request.POST)
+    data = json.loads(response_json)
+    user = request.user
+    if user.email == data['email']:
+        error_message = 'It is your existing email address'
+        messages.error(request, error_message)
+    else:
+        user.email = data['email']
+        user.save()
+        success_message = f'You changed your email. Every messages will be send to \
+                    {request.user.email} until now.'
+        messages.success(request, success_message)
+
+
+class SiteDownChecker:
+
+    def __init__(self, url, user):
+        self.url = url
+        self.time = 0
+        self.error = None
+        self.user = user
+
+    def status(self, proxy=False):
+        try:
+            if proxy:
+                r = ProxyRequests(self.url)
+            else:
+                r = requests.get(self.url, headers={
```

Comment:
1) Zrób nową linię po `.get(` będzie ładniej wyglądało. 2) Czy na pewno potrzebujesz wysyłać te headery? Mówisz w nich że jesteś mozillą. Czy to ma znaczenie tutaj?